automol/util/vector.py

```diff
@@ -1,65 +1,64 @@
-""" vector functions
-"""
-from typing import Callable, Optional, Tuple
+"""vector functions."""
+from collections.abc import Callable, Sequence
 
-import numpy
+import numpy as np
 from scipy.spatial.transform import Rotation
+
 from phydat import phycon
 
-Vector = Tuple[float, float, float]
+Vector = tuple[float, float, float]
 
 
-def unit_norm(xyz):
+def unit_norm(xyz: Sequence | np.ndarray) -> float:
     """Normalize a vector (xyz) to 1.0.
 
     Returns 0. for null vectors
 
-    :param xyz: vector
-    :type xyz: tuple, list, or numpy nd.array
-    :rtype: float
+    :param xyz: Vector
+    :return: Numbers in the vector
     """
     uxyz = xyz
-    norm = numpy.linalg.norm(xyz)
-    if not numpy.isclose(norm, 0.):
-        uxyz = numpy.divide(xyz, norm)
+    norm = np.linalg.norm(xyz)
+    if not np.isclose(norm, 0.0):
+        uxyz = np.divide(xyz, norm)
     return uxyz
 
 
-def unit_direction(xyz1, xyz2):
+def unit_direction(xyz1: Sequence | np.ndarray, xyz2: Sequence | np.ndarray) -> float:
     """Calculate a unit direction vector from `xyz1` to `xyz2`.
 
     :param xyz1: 3D vector
-    :type xyz1: tuple, list, or numpy nd.array
     :param xyz2: 3D vector
-    :type xyz2: tuple, list, or numpy nd.array
-    :rtype: float
+    :return: Unit direction vector from two vectors
     """
-    dxyz12 = numpy.subtract(xyz2, xyz1)
+    dxyz12 = np.subtract(xyz2, xyz1)
     uxyz12 = unit_norm(dxyz12)
     return uxyz12
 
 
-def are_parallel(xyz1, xyz2, orig_xyz=(0.0, 0.0, 0.0), anti=False, tol=1e-3):
+def are_parallel(
+    xyz1: Sequence | np.ndarray,
+    xyz2: Sequence | np.ndarray,
+    orig_xyz: Sequence | np.ndarray = (0.0, 0.0, 0.0),
+    anti: bool = False,
+    tol: float = 1e-3,
+) -> bool:
     """Assess if two vectors are parallel to each other.
 
     :param xyz1: 3D vector
-    :type xyz1: tuple, list, or numpy nd.array
     :param xyz2: 3D vector
-    :type xyz2: tuple, list, or numpy nd.array
-    :param orig_xyz: origin of coordinate system `xyz1` and `xyz2` are in
-    :type orig_xyz: tuple, list, or numpy nd.array
+    :param orig_xyz: Origin of coordinate system `xyz1` and `xyz2` are in
     :param anti: Test for anti-parallelity too?
-    :type anti: bool
-    :param tol: tolerance for checking determinant
-    :type tol: float
-    :rtype: bool
+
+    :param tol: Tolerance for checking determinant
+    :return: True if vectors are parallel, false if not
     """
```

Comment:
Again, we will need to call the `require_vector_like` function for `xyz1`, `xyz2`, and `orig_xyz`.